src/components/atoms/InputField/InputField.scss

```diff
@@ -0,0 +1,51 @@
+@import "scss/constants";
+
+$icon-width: 2.5rem;
+
+.input-field {
+  position: relative;
+
+  &--icon-start {
+    input {
+      padding-left: $icon-width;
+    }
+  }
+
+  &--icon-end {
+    input {
+      padding-right: $icon-width;
+    }
+  }
+
+  &__input {
+    height: 100%;
+
+    &:focus + .input-field__icon > * {
+      opacity: 0.6;
+    }
```

Comment:
I don't really understand the logic of this.
But is it possible to not use the wildcard selector(> *)?
Given that we assign opacity, you can assign it to a parent(and it will affect he children)

Reply:
I used wildcard selector for custom icon elements which we can pass as `iconStart`/`iconEnd`. Let's say we use img-tag as `iconStart`:

```tsx
iconStart={<img
  className="nav__clear-search"
  src="/icons/nav-dropdown-close.png"
  alt="close button"
  onClick={clearSearchQuery}
/>}
```

so we can now overwrite opacity for this icon and add its own styles using `.nav__clear-search` selector. But if we assign opacity to parent element instead, we'll have to go inside `InputField` component searching for the className (`.input-field__icon`) that we need to overwrite. I thought that it may be uncomfortable to do that every time and also, it produces selector `.input-field__icon` which can't be found directly on the page where we use `InputField`, so someone can accidentally remove it thinking that it's not needed here.

so what do you think? should still I remove wildcard selector?

Reply:
I see. Makes sense, thanks for explaining